Cannot find module './config/environment'

With a new application, TypeScript cannot find the config;
```
# app/router.ts
import config from './config/environment'
```

To reproduce:
```
ember new ember-ts
cd ember-ts
ember install ember-cli-typescript
mv app/router.js app/router.ts
ember build
```

This gives the following error:
```
app/router.ts(2,20): error TS2307: Cannot find module './config/environment'.
```

That produces the following error -- still unable to find the type definitions:
```
12:50:54 PM - Starting compilation in watch mode...
app/router.ts(2,20): error TS2307: Cannot find module 'ember-ts/config/environment'.
12:50:59 PM - Compilation complete. Watching for file changes.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) a
 ### Fixed
 
 * Addon components need to manually set their layout property to the imported compiled template.
+* The declaration file for the `<app-name>/config/environment` module now resolves correctly from app code. If you have a version of this file previously generated at `types/<app-name>/config/environment.d.ts`, you'll likely want to move it to `app/config/environment.d.ts`.
 
 ## [1.1.6] - 2018-02-23
 

--- a/README.md
+++ b/README.md
@@ -61,7 +61,7 @@ In addition to ember-cli-typescript, we make the following changes to your proje
 
   * [`tsconfig.json`](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html)
   * `types/<app name>/index.d.ts` – the location for any global type declarations you need to write for you own application; see [Global types for your package](#global-types-for-your-package) for information on its default contents and how to use it effectively
-  * `types/<app name>/config/environment.d.ts` – a basic set of types defined for the contents of the `config/environment.js` file in your app; see [Environment and configuration typings](#environment-and-configuration-typings) for details
+  * `app/config/environment.d.ts` – a basic set of types defined for the contents of the `config/environment.js` file in your app; see [Environment and configuration typings](#environment-and-configuration-typings) for details
 
 ### Ember support
 
@@ -147,9 +147,9 @@ In the case of applications (but not for addons), we also automatically include 
 
 #### Environment configuration typings
 
-Along with the @types/ files mentioned above, ember-cli-typescript adds a starter interface for `config/environment.js` in `config/environment.d.ts`. This interface will likely require some changes to match your app.
+Along with the @types/ files mentioned above, ember-cli-typescript adds a starter interface for `config/environment.js` in `app/config/environment.d.ts`. This interface will likely require some changes to match your app.
 
-We install this file because the actual `config/environment.js` is (a) not actually identical with the types as you inherit them in the content of an application, but rather a superset of what an application has access to, and (b) not in a the same location as the path at which you look it up. We map it to the lookup path within your `types` directory, and TypeScript resolves it correctly.
+We install this file because the actual `config/environment.js` is (a) not actually identical with the types as you inherit them in the content of an application, but rather a superset of what an application has access to, and (b) not in a the same location as the path at which you look it up. The actual `config/environment.js` file executes in Node during the build, and Ember CLI writes its result as `<my-app>/config/environment` into your build for consumption at runtime.
 
 ### String-keyed lookups
 

--- a/blueprints/ember-cli-typescript/files/__root__/config/environment.d.ts
+++ b/blueprints/ember-cli-typescript/files/__root__/config/environment.d.ts
@@ -7,10 +7,10 @@ export default config;
  * For now these need to be managed by the developer
  * since different ember addons can materialize new entries.
  */
-declare namespace config {
-  var environment: any;
-  var modulePrefix: string;
-  var podModulePrefix: string;
-  var locationType: string;
-  var rootURL: string;
-}
+declare const config: {
+  environment: any;
+  modulePrefix: string;
+  podModulePrefix: string;
+  locationType: string;
+  rootURL: string;
+};

--- a/node-tests/blueprints/ember-cli-typescript-test.js
+++ b/node-tests/blueprints/ember-cli-typescript-test.js
@@ -77,7 +77,7 @@ describe('Acceptance: ember-cli-typescript generator', function() {
         expect(projectTypes).to.exist;
         expect(projectTypes).to.include(ects.APP_DECLARATIONS);
 
-        const environmentTypes = file('types/my-app/config/environment.d.ts');
+        const environmentTypes = file('app/config/environment.d.ts');
         expect(environmentTypes).to.exist;
 
         const emberDataCatchallTypes = file('types/ember-data.d.ts');
@@ -121,6 +121,9 @@ describe('Acceptance: ember-cli-typescript generator', function() {
         const projectTypes = file('types/dummy/index.d.ts');
         expect(projectTypes).to.exist;
         expect(projectTypes).not.to.include(ects.APP_DECLARATIONS);
+
+        const environmentTypes = file('tests/dummy/app/config/environment.d.ts');
+        expect(environmentTypes).to.exist;
 
         const emberDataCatchallTypes = file('types/ember-data.d.ts');
         expect(emberDataCatchallTypes).not.to.exist;

